src/CodeGen_LLVM.cpp

```diff
@@ -1472,6 +1472,14 @@ Value *CodeGen_LLVM::codegen(const Expr &e) {
     value = nullptr;
     e.accept(this);
     internal_assert(value) << "Codegen of an expr did not produce an llvm value\n";
+
+    // Halide doesn't distinguish between scalars and vectors of size 1.
```

Comment:
might be good to expand this comment: the code below is converting a vector of size 1 to a scalar?